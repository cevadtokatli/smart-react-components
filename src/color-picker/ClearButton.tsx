import React from "react"
import {SetValue$} from "../types/color-picker"
import * as CT from "../constants/color-picker/change-type"
import ConsoleButtonElement from "./ConsoleButtonElement"
import ClearIcon from "./icons/Clear"

interface Props {
    setValue: SetValue$
}

const ClearButton: React.FC<Props> = ({setValue}) => {
    const setColor = () => setValue(null, null, CT.BUTTON)

    return (
        <ConsoleButtonElement onClick={setColor}>
            <ClearIcon />
        </ConsoleButtonElement>
    )
}

export default ClearButton