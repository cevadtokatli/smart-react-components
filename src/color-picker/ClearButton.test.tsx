import React from "react"
import renderTheme from "../test/renderTheme"
import mountTheme from "../test/mountTheme"
import ClearButton from "./ClearButton"

describe("<ClearButton />", () => {
    let props

    beforeEach(() => {
        props = {
            setValue: jest.fn()
        }
    })

    it("should match snapshot", () => {
        const wrapper = renderTheme(<ClearButton {...props} />)
        expect(wrapper).toMatchSnapshot()
    })

    it("should call setValue", () => {
        const wrapper = mountTheme(<ClearButton {...props} />)
        wrapper.find("div").simulate("click")
        expect(props.setValue).toHaveBeenCalled()
    })
})