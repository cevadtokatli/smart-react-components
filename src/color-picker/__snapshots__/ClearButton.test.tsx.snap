// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClearButton /> should match snapshot 1`] = `
<div
  class="sc-AxjAm sc-AxirZ dSFjTy"
>
  <svg
    class="sc-AxiKw sc-AxhCb lnpmyP"
    viewBox="0 0 2048 1792"
  >
    <path
      d="M960 1408l336-384h-768l-336 384h768zm1013-1077q15 34 9.5 71.5t-30.5 65.5l-896 1024q-38 44-96 44h-768q-38 0-69.5-20.5t-47.5-54.5q-15-34-9.5-71.5t30.5-65.5l896-1024q38-44 96-44h768q38 0 69.5 20.5t47.5 54.5z"
    />
  </svg>
</div>
`;
